src/main/java/org/broadinstitute/hellbender/tools/spark/utils/HopscotchCollection.java

```diff
@@ -562,6 +557,25 @@ public T next() {
             final T result = buckets[currentIndex];
             removeIndex = currentIndex;
             removePrevIndex = prevIndex;
+            advanceUntilEquivalent();
+            return result;
+        }
+
+        @Override
+        public void remove() {
+            super.remove();
+            if ( currentIndex != NO_ELEMENT_INDEX ) {
+                checkEquivalence();
+            }
+        }
+
+        private void checkEquivalence() {
```

Comment:
I agree with Chris on this one.
What I've observed (noticing that this is my first time looking at this code):
  * `checkEquivalence()` essentially adds a `while` check that also appears in `advanceUntilEquivalent()`, which is its delegate, so `checkEquivalence()` could be removed with a re-structured `advanceUntilEquivalent()`
  * `advanceUntilEquivalent()` could be&mdash;but not necessary&mdash;renamed `advanceUntilEquivalentOrExhausted()`